### PR TITLE
fix #2506 Switch to GitHub Actions for CI (PR checks)

### LINKED ISSUE #2506: Switch to GitHub Actions for CI
See reactor/reactor#690 for context.

For core, we can use a matrix of jobs to execute some blocks in parallel: core tests, jcstress and non-core tests.

see #2508 for details of out-of-code steps

### PR DESCRIPTION
**Pull request by simonbasle**
- [x] validate the workflow
 - [x] remove `travis.yml`
 - [ ] exclude this repo from scope of TravisCI app in settings
 - [x] make the newly introduced checks `required` in settings

### CODE REVIEW

File: .github/workflows/pr-check.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  gradle:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 8
+    - uses: eskatos/gradle-command-action@v1
+      with:
+        arguments: check
```
Comment (ericbottard) [inline .github/workflows/pr-check.yml:13]:
```suggestion
        arguments: check --no-daemon
        wrapper-cache-enabled: true
        dependencies-cache-enabled: true
        configuration-cache-enabled: true
```

The third cache line doesn't bring much, but doesn't hurt either.

File: .github/workflows/pr-check.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+name: PR Check
+on: pull_request
+jobs:
+  gradle:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
```
Comment (ericbottard) [inline .github/workflows/pr-check.yml:7]: Could add `name:` for readability of steps

**Review by ericbottard — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-check.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+name: PR Check
```
Comment (ericbottard) [inline .github/workflows/pr-check.yml:1]: Could rename to build & test on also apply on main branches (with `on: push`)

Comment (simonbasle) [reply .github/workflows/pr-check.yml:1]: we don't want to build snapshots of PR branches though...

Comment (ericbottard) [reply .github/workflows/pr-check.yml:1]: this is why I added "on main branches". You can limit those, see eg https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags

Comment (simonbasle) [reply .github/workflows/pr-check.yml:1]: that's by workflow. how would you exclude the job/step of building a pr commit? isn't it far simpler to introduce a separate workflow on:push which targets maintenance branches/master, that will effectively test+build+publish snapshots?

Comment (ericbottard) [reply .github/workflows/pr-check.yml:1]: Ah, you're right, once we factor in the publication step, we're better off with a different workflow indeed. Sorry for the noise.

**Review by aneveu — APPROVED** [review]

**Review by aneveu — APPROVED** [review]

**Comment by simonbasle** [comment]
rebased and reworked into 2 meaningful commits to be `Rebase and merge`, ready for final review @aneveu @ericbottard

**Review by aneveu — APPROVED** [review]

**Pull request merged**